flowcraft/tests/template_tests/test_mapping2json.py

```diff
@@ -0,0 +1,120 @@
+import os
+import json
+import pytest
+import flowcraft.templates.mapping2json as mapping2json
+
+depth_dict_coverage = {
+    "ACC1": {
+        "1": 20,
+        "2": 30,
+        "3": 50,
+        "4": 40,
+        "5": 30,
+        "6": 20,
+        "7": 20,
+        "8": 40,
+        "9": 30,
+        "10": 20
+    },
+    "ACC2": {
+        "1": 20,
+        "2": 30,
+        "3": 50,
+        "4": 40,
+        "7": 20,
+        "8": 40,
+        "9": 30,
+        "10": 20
+    }
+}
+
+#
+plasmid_length = {
+    "ACC1": "10",
+    "ACC2": "10"
+}
+
+
+@pytest.fixture
+def fetch_file(tmpdir, request):
+    # create a temporary file depth txt file
+    depth_file = tmpdir.join("test_depth_file.txt")
+    depth_file.write("ACC1\t1\t30")
+    # creates a temporary file with the json_dict
+    json_dict = tmpdir.join("test_json_dict.json")
+    json_dict.write('{"ACC1": 2000}')
+    # executes the function to be tested
+    mapping2json.main(str(depth_file), str(json_dict), "0", "test")
+    result_dict = json.load(open("{}_mapping.json".format(str(depth_file))))
+
+    # finalizer statement that removes .report.json
+    def remove_test_files():
+        os.remove(".report.json")
+    request.addfinalizer(remove_test_files)
+
+    return result_dict, str(depth_file)
+
+
+def test_depth_file_reader(tmpdir):
+    """
+    test the output of depth_file_reader_function to be a given dict
+    """
+
+    # create a temporary file to make a dict from
+    file = tmpdir.join("test_depth_file.txt")
```

Comment:
The use of the `file` variable is not recommended. It was a builtin in py2 and occasionally used for type comparisons in py3. You can even notice it has a different syntax highlight

```suggestion
    file_handle = tmpdir.join("test_depth_file.txt")
```